### PR TITLE
Add random nonces to Oak C++ and Java Crypto

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR makes Oak C++ and Java Hybrid Encryption use random nonces.

Ref https://github.com/project-oak/oak/issues/4507

### CODE REVIEW

File: cc/crypto/hpke/utils.cc [hunk 1]
```diff
@@ -63,34 +64,10 @@ absl::StatusOr<std::unique_ptr<EVP_AEAD_CTX>> GetContext(EVP_HPKE_CTX* hpke_ctx,
...
-  // We use 8 here since sequence number is 64 bits.
-  for (size_t i = 0; i < 8; ++i) {
-    // Get the first 8 bits and push bits right since the encoded nonce is big-endian.
-    nonce[kAeadNonceSizeBytes - i - 1] = sequence_number & 0xff;
-    sequence_number >>= 8;
-  }
-  // XOR each of the nonce bits with the base nonce.
   for (size_t i = 0; i < kAeadNonceSizeBytes; ++i) {
-    nonce[i] ^= base_nonce.at(i);
+    nonce[kAeadNonceSizeBytes - i - 1] = rand() & 0xff;
```
Comment (conradgrobler) [inline cc/crypto/hpke/utils.cc:70]: This is not a cryptographically secure way to generate random nonces. This should use random number generation primitives from boringssl (https://commondatastorage.googleapis.com/chromium-boringssl-docs/rand.h.html#RAND_bytes).

Comment (ipetr0v) [reply cc/crypto/hpke/utils.cc]: Done

File: cc/crypto/hpke/utils.cc [hunk 2]
```diff
@@ -57,40 +58,16 @@ absl::StatusOr<std::unique_ptr<EVP_AEAD_CTX>> GetContext(EVP_HPKE_CTX* hpke_ctx,
...
-  // We use 8 here since sequence number is 64 bits.
-  for (size_t i = 0; i < 8; ++i) {
-    // Get the first 8 bits and push bits right since the encoded nonce is big-endian.
-    nonce[kAeadNonceSizeBytes - i - 1] = sequence_number & 0xff;
-    sequence_number >>= 8;
-  }
-  // XOR each of the nonce bits with the base nonce.
-  for (size_t i = 0; i < kAeadNonceSizeBytes; ++i) {
-    nonce[i] ^= base_nonce.at(i);
+  if (!RAND_bytes(&nonce[0], nonce.size())) {
```
Comment (conradgrobler) [inline cc/crypto/hpke/utils.cc:69]: I think `nonce.data()` is more idiomatic than `&nonce[0]` to get a pointer to the start of the underlying buffer.

**Review by conradgrobler — APPROVED** [review]
Thanks.

File: cc/crypto/hpke/utils.cc (hunk 2 above)
Comment (ipetr0v) [reply cc/crypto/hpke/utils.cc]: Done

**Pull request merged**